src/main/java/com/google/sps/login/Login.java

```diff
@@ -33,35 +33,37 @@
     private static final String REDIRECT_LOGIN = "/";
     private static final String REDIRECT_LOGOUT = "/";
     
-    // Print json of whether user is logged in and the respective login/logout link
+    // Passes in the actual userService to getAuthentication
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-        Authentication auth;
+        UserService userService = UserServiceFactory.getUserService();
         
-        response.setContentType("application/json");
+        // Get the login status and link
+        Authentication auth = getAuthentication(userService);
+
+        Gson gson = new Gson();
+        String json = gson.toJson(auth);
+
+        response.getWriter().println(json);
```

Comment:
Can you inline lines 45+46 with this?  seems like it can just be println(new Gson().toJson(auth))